### DISCUSSION TITLE
Could not find a declaration file for module 'ws'

### DISCUSSION THREAD

**Discussion by benesjan** [wevm/viem#203] [Question]
I am trying to use viem in combination with node v18.15.0, newest viem (0.1.14) and ESM and I am getting the following error:

> ../.yarn/__virtual__/isomorphic-ws-virtual-4abf445193/0/cache/isomorphic-ws-npm-5.0.0-9c72251c2d-e20eb2aee0.zip/node_modules/isomorphic-ws/index.d.ts(6,28): error TS7016: Could not find a declaration file for module 'ws'. '/Users/jan/Coding/aztec3-packages/yarn-project/.yarn/__virtual__/ws-virtual-26db9323a7/0/cache/ws-npm-8.13.0-26ffa3016a-53e991bbf9.zip/node_modules/ws/index.js' implicitly has an 'any' type.
>   Try `npm i --save-dev @types/ws` if it exists or add a new declaration (.d.ts) file containing `declare module 'ws';`

I tried installing the types with `yarn add -D @types/ws` but it didn't help.

Any ideas what is wrong?

**Comment by jxom** [comment]
Any chance you could upload a minimal reproduction via a repo?

**Reply by benesjan** [reply]
Hello, I have not yet managed to reproduce it in a separate minimal repo. I think the issue has to do something with yarn PNP. We use that in our main repository at Aztec. I will report back if I manage to identify it and I will close the discussion if it will simply turn out to be some misconfiguration on our side (which is most likely the case).

**Comment by benesjan** [comment]
After switching away from `yarn PNP` the issue resolved itself.
